exonum/benches/criterion/storage.rs

```diff
@@ -45,58 +52,119 @@ fn generate_random_kv(len: usize) -> Vec<([u8; KEY_SIZE], Vec<u8>)> {
             rng.fill_bytes(&mut k);
         }
         exists_keys.insert(k);
-        (k, v)
+        (Hash::new(k), v)
     };
 
     (0..len).map(kv_generator).collect::<Vec<_>>()
 }
 
-fn merkle_table_insertion<T: Database>(b: &mut Bencher, db: &T) {
+fn proof_list_append<T: Database>(b: &mut Bencher, db: &T, len: &usize) {
     let mut rng = XorShiftRng::from_seed([200; 16]);
     let mut storage = db.fork();
     let mut table = ProofListIndex::new(NAME, &mut storage);
 
-    b.iter(|| {
-        let v_generator = |_| {
-            let mut chunk = vec![0; 10];
-            rng.fill_bytes(&mut chunk);
-            chunk
-        };
-
-        for item in (0..1000).map(v_generator) {
-            table.push(item);
-        }
-    });
+    b.iter_with_setup(
+        || {
+            (0..*len)
+                .map(|_| {
+                    let mut chunk = vec![0; 64];
+                    rng.fill_bytes(&mut chunk);
+                    chunk
+                }).collect::<Vec<_>>()
+        },
+        |data| {
+            for item in data {
```

Comment:
Just to be clear: the table *is not* cleared on each iteration, is it? Is this intended? I *could* let it slide in case of `ProofMapIndex`, but in case of lists, not clearing the list makes the performance of each successive iteration increasingly worse.

Reply:
No, in each setup I create a new empty fork and wouldn't merge it after bench iteration.

Reply:
The setup closure is just above in the code, and the fork isn't created there (but rather earlier) as of 275ef0c526ae90670ee7a8c57e64b8c9a9570be1 😕 If you add `assert!(table.is_empty())` before the `for` cycle, it will fail.

Reply:
`assert!(table.keys().next().is_none());` doesn't fail.